vm/http_client_test.go

```diff
@@ -0,0 +1,96 @@
+package vm
+
+import "testing"
+
+func TestHTTPClientObject(t *testing.T) {
+
+	//blocking channel
+	c := make(chan bool, 1)
+
+	//server to test off of
+	go startTestServer(c)
+
+	tests := []struct {
+		input    string
+		expected interface{}
+	}{
+		//test get request
+		{`
+		require "net/http"
+
+		c = Net::HTTP::Client.new
+
+		res = c.send do |req|
+			req.url = "http://127.0.0.1:3000/index"
+			req.method = "GET"
+		end
+
+		res.body
+		`, "GET Hello World"},
+		{`
+		require "net/http"
+
+		c = Net::HTTP::Client.new
+
+		res = c.send do |req|
+			req.url = "http://127.0.0.1:3000/index"
+			req.method = "POST"
+			req.body = "Hi Again"
+		end
+
+		res.body
+		`, "POST Hi Again"},
+		{`
+		require "net/http"
+
+		c = Net::HTTP::Client.new
+
+		res = c.send do |req|
+			req.url = "http://127.0.0.1:3000/error"
+			req.method = "GET"
+		end
+
+		res.status_code
+		`, 404},
+	}
+
+	//block until server is ready
+	<-c
+
+	for i, tt := range tests {
+		v := initTestVM()
+		evaluated := v.testEval(t, tt.input, getFilename())
+		checkExpected(t, i, evaluated, tt.expected)
+		v.checkCFP(t, i, 0)
+		v.checkSP(t, i, 1)
+	}
+}
+
+func TestHTTPClientObjectFail(t *testing.T) {
+
+	testsFail := []errorTestCase{
+		{`
+		require "net/http"
+
+		c = Net::HTTP::Client.new
+
+		res = c.send do |req|
+			req.url = "http://127.0.0.1:3001"
+			req.method = "GET"
+		end
+
+		res
+		`, "InternalError: Could not get response: Get http://127.0.0.1:3001: dial tcp 127.0.0.1:3001: getsockopt: connection refused", 6},
```

Comment:
I thought you said you will add `HTTPError`?

Reply:
And this message has duplicated informations. If we want to make users know what's going on in our way, we shouldn't still pass the message we get from Go.

Reply:
And this message has duplicated informations. If we want to make users know what's going on in our way, we shouldn't still pass the message we get from Go.